src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/MinioIntegrationTest.java

```diff
@@ -0,0 +1,181 @@
+        minioServer.start();
+
+        Integer mappedPort = minioServer.getFirstMappedPort();
+        Testcontainers.exposeHostPorts(mappedPort);
+        minioServiceEndpoint = String.format("%s:%s", minioServer.getContainerIpAddress(), mappedPort);
+        
+        image = ArtifactManagerTest.prepareImage();
+    }
+    
+    @AfterClass
+    public static void shutDownClass() {
+        if (minioServer.isRunning()) {
+            minioServer.stop();
+        }
+    }
+    
+    @Before
+    public void setUp() throws IOException {
+        provider = new S3BlobStore();
+        CredentialsAwsGlobalConfiguration credentialsConfig = CredentialsAwsGlobalConfiguration.get();
+        credentialsConfig.setRegion(REGION);
+        CredentialsProvider.lookupStores(jenkinsRule.jenkins)
+                .iterator()
+                .next()
+                .addCredentials(Domain.global(), new AWSCredentialsImpl(CredentialsScope.GLOBAL, "MinioIntegrationTest", ACCESS_KEY, SECRET_KEY, "MinioIntegrationTest"));
+        credentialsConfig.setCredentialsId("MinioIntegrationTest");
+        
+        config = S3BlobStoreConfig.get();
+        config.setContainer(CONTAINER_NAME);
+        config.setPrefix(CONTAINER_PREFIX);
+        config.setCustomEndpoint(minioServiceEndpoint);
+        config.setUseHttp(true);
+        config.setUsePathStyleUrl(true);
+        config.setDisableSessionToken(true);
+        client = config.getAmazonS3ClientBuilderWithCredentials().build();
+        artifactManagerFactory = new JCloudsArtifactManagerFactory(provider);
+    }
+    
+    private void createBucketWithAwsClient(String bucketName) {
+        config.setContainer(bucketName);
+        client.createBucket(bucketName);
+    }
+    
+    @Test
+    public void canCreateBucket() throws Exception {
+        String testBucketName = "jenkins-ci-data";
+        Bucket createdBucket = config.createS3Bucket(testBucketName);
+        assertEquals(testBucketName, createdBucket.getName());
+        client.headBucket(new HeadBucketRequest(testBucketName));
+    }
+    
+    @Test
+    public void artifactArchive() throws Exception {
+        createBucketWithAwsClient("artifact-archive");
+        ArtifactManagerTest.artifactArchive(jenkinsRule, artifactManagerFactory, true, image);
+    }
+
+    @Test
+    public void artifactArchiveAndDelete() throws Exception {
+        createBucketWithAwsClient("artifact-archive-and-delete");
+        ReflectionTestUtils.setField(config, "deleteArtifacts", true);
+        ArtifactManagerTest.artifactArchiveAndDelete(jenkinsRule, artifactManagerFactory, true, image);
+    }
+    
+    @Test//(expected = AssertionError.class)
+    public void artifactArchiveAndDeleteShouldFail() throws Exception {
+        // unfortunately, we don't get a message to verify against
+        // so we are just verifying that it fails with an AssertionError
+        expectedException.expect(AssertionError.class);
```

Comment:
Remove this. I do not get why this is here. The test fails

```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.jenkinsci.plugins.workflow.ArtifactManagerTest.lambda$artifactArchiveAndDelete$1(ArtifactManagerTest.java:158)
	at org.jenkinsci.plugins.workflow.ArtifactManagerTest.wrapInContainer(ArtifactManagerTest.java:125)
	at org.jenkinsci.plugins.workflow.ArtifactManagerTest.artifactArchiveAndDelete(ArtifactManagerTest.java:154)
	at io.jenkins.plugins.artifact_manager_jclouds.s3.MinioIntegrationTest.artifactArchiveAndDeleteShouldFail(MinioIntegrationTest.java:153)
```

which seems like a bug, in the test at least.

Reply:
Ok I'll remove this